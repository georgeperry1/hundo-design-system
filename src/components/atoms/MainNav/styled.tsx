import styled from 'styled-components';

export const MainNavContainer = styled.div`
  height: 100vh;
  background: #111111;
  width: 220px;
  padding: 35px 0 180px;
`;

export const MainNavWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const LogoImage = styled.img`
  margin-bottom: 50px;
`;
