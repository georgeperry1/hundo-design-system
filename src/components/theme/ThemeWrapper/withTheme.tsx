import React from 'react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../themes';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
  const ComponentWithTheme = (props: P) => {
    return (
      <ThemeProvider theme={lightTheme}>
        <WrappedComponent {...props} />
      </ThemeProvider>
    );
  };
  return ComponentWithTheme;
}
export default withTheme;
