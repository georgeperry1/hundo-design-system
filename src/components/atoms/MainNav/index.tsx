import React, { ReactChild } from 'react';
import Logo from '/stories/assets/hundo-light.svg';

import { MainNavContainer, MainNavWrapper, LogoImage } from './styled';

const MainNav: React.FC = ({ children }) => (
  <MainNavContainer>
    <MainNavWrapper>
      <LogoImage src={Logo} />
      {children}
    </MainNavWrapper>
  </MainNavContainer>
);

export default MainNav;
