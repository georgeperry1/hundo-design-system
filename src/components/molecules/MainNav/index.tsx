import React, { Component } from 'react';

import { MainNavContainer, MainNavWrapper, LogoWrapper } from './styled';

type MainNavProps = {
  Logo?: typeof Component;
};

const MainNav: React.FC<React.PropsWithChildren<MainNavProps>> = ({ Logo, children }) => (
  <MainNavContainer>
    <MainNavWrapper>
      {Logo && (
        <LogoWrapper>
          <Logo />
        </LogoWrapper>
      )}
      {children}
    </MainNavWrapper>
  </MainNavContainer>
);

export default MainNav;
