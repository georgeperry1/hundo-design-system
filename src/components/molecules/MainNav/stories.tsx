import { Story, Meta } from '@storybook/react';

import MainNav from '.';
import NavItem from '../../atoms/MainNavItem';

export default {
  title: 'Components/Molecules/MainNav',
  component: MainNav,
} as Meta;

const Template: Story = (args) => (
  <MainNav {...args}>
    <NavItem {...args}>{args.text}</NavItem>
  </MainNav>
);

export const Default = Template.bind({});

Default.args = {
  active: false,
  text: 'Profile',
  icon: 'https://www.svgrepo.com/show/794/male-user.svg',
  backgroundColor: '#FFFF50',
  shadowColor: '#B7B73D',
};

export const WithLogo = Template.bind({});

WithLogo.args = {
  active: false,
  text: 'Profile',
  icon: 'https://www.svgrepo.com/show/794/male-user.svg',
  backgroundColor: '#FFFF50',
  shadowColor: '#B7B73D',
  logo: 'https://www.svgrepo.com/show/794/male-user.svg',
};
