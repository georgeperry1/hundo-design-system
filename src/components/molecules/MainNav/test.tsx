import { render } from '@testing-library/react';

import MainNav from '.';
import ThemeWrapper from '../../../utils/test-theme-wrapper';

describe('MainNav', () => {
  it('should render', () => {
    const { baseElement } = render(
      <ThemeWrapper
        TestedComponent={MainNav}
        componentProps={{ logo: 'https://www.svgrepo.com/show/794/male-user.svg' }}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
