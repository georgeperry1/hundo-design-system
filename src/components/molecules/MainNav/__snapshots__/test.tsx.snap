// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav should render 1`] = `
.c0 {
  height: 100vh;
  background: #111111;
  width: 220px;
  padding: 35px 0 180px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  margin-bottom: 50px;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div>
            <div>
              <div
                data-src="../../../assets/hundo-light.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
